tests/workflows/test_optuna.py

```diff
@@ -0,0 +1,93 @@
+import uuid
+
+import coiled
+import distributed
+import optuna
+import pytest
+import xgboost as xgb
+from dask.distributed import Client, wait
+from optuna.integration.dask import DaskStorage
+from optuna.samplers import RandomSampler
+from packaging.version import Version
+from sklearn.datasets import fetch_covtype
+from sklearn.model_selection import KFold, cross_val_score
+from sklearn.preprocessing import LabelEncoder
+
+
+@pytest.fixture(scope="module")
+def optuna_cluster(
+    dask_env_variables,
+    cluster_kwargs,
+    github_cluster_tags,
+):
+    with coiled.Cluster(
+        f"optuna-{uuid.uuid4().hex[:8]}",
+        environ=dask_env_variables,
+        tags=github_cluster_tags,
+        **cluster_kwargs["optuna_cluster"],
+    ) as cluster:
+        yield cluster
+
+
+@pytest.fixture
+def optuna_client(
+    optuna_cluster,
+    cluster_kwargs,
+    upload_cluster_dump,
+    benchmark_all,
+):
+    n_workers = cluster_kwargs["optuna_cluster"]["n_workers"]
+    with Client(optuna_cluster) as client:
+        optuna_cluster.scale(n_workers)
+        client.wait_for_workers(n_workers)
+        client.restart()
+        with upload_cluster_dump(client), benchmark_all(client):
+            yield client
+
+
+@pytest.mark.skipif(
+    Version(distributed.__version__) >= Version("2023.3.2")
+    # Use `.base_version` to account for `optuna` `main` branch
+    and Version(Version(optuna.__version__).base_version) < Version("3.2.0"),
+    reason=(
+        "There was a change in distributed that broke the dask + optuna integration. "
+        "A fix will be included in optuna=3.2.0."
+    ),
+)
+def test_optuna(optuna_client):
+    # We use a random sampler with a seed to get deterministic results.
+    # This is just for benchmarking purposes.
+    study = optuna.create_study(
+        direction="maximize", storage=DaskStorage(), sampler=RandomSampler(seed=2)
+    )
+
+    def objective(trial):
+        # Dataset (241.59 MiB) from http://archive.ics.uci.edu/ml/datasets/covertype
+        X, y = fetch_covtype(return_X_y=True)
+
+        # Format training labels
+        le = LabelEncoder()
+        y = le.fit_transform(y)
+
+        # Get hyperparameter values for this trial and model score
+        params = {
+            "n_estimators": trial.suggest_int("n_estimators", 2, 10),
+            "max_depth": trial.suggest_int("max_depth", 2, 10),
+            "learning_rate": trial.suggest_float("learning_rate", 1e-8, 1.0, log=True),
+            "subsample": trial.suggest_float("subsample", 0.2, 1.0),
+            "eval_metric": "mlogloss",
```

Comment:
Is there a reason we are going off the default here? I'm not super familiarized with the problem, I just know the default for the Classifier is `logloss`. I was looking in the [docs](https://xgboost.readthedocs.io/en/stable/parameter.html) what `mlogloss` was but it links to something on sklearn so I got confused

Reply:
I initially chose `mlogloss` because this isn't a binary classification problem, but yeah, I think you're right that the default `logloss` is probably more appropriate